### ISSUE TITLE
Chrome on Android: Touching outside already opened dropdown doesn't close it back

### ISSUE THREAD

**Issue by shehi** [snapappointments/bootstrap-select#1558]
Hi,

As you know, in Desktop versions of web browsers, when one opens the dropdown and then clicks anywhere on the page, outside the dropdown, that closes the dropdown.

However, that's not the case in Chrome on Android: touching anywhere on the page, outside the dropdown, doesn't close it. You get stuck with opened dropdown, unless you touch the previous value and make it close. This is quite frustrating experience.

I believe this may have to do with possible missing touch-event handlers in this package. Hope you guys fix it soon. 

Thanks for this awesome tool by the way!

**Comment by shehi** [comment]
I just checked this very situation in Firefox on Android N as well, the same behavior is observed.

As a side note: when a dropdown is open, it also blocks other dropdowns from opening (guess, it requires the previous to close first). Continuously trying to open some other dropdown menu while there is already one open, doesn't do anything.

**Comment by birjolaxew** [comment]
Cannot reproduce on [examples page](https://silviomoreto.github.io/bootstrap-select/examples/).
Chrome: 54.0.2840.85
Android: 6.0.1

**Comment by shehi** [comment]
Example pages use jQuery 1.11.3. I tested with jQuery 2.2.4. And not to mention, we have jQuery 3 nowadays, iirc.

**Comment by shehi** [comment]
Couldn't find Android 6 or 7 on SauceLabs, but Android 5 is consistent: Examples page still works whereas my implementation doesn't:

https://saucelabs.com/beta/tests/6b8d8acccce24ac4a7bf14fca5d6416c - my implementation
https://saucelabs.com/beta/tests/b3f7478f63b84fb3bb1e22945a0f1379 - examples page

You can see the URL of my implementation (with HTTP auth pass included). Could you please have a look and let me know what I did wrong? (I have JS assets merged with a bundler, so you will find a single JS file with everything in it). Page is very clean, with semantic markup.

**Comment by birjolaxew** [comment]
@shehi Thanks for the extra info.

I've created a [minimal test case](https://jsfiddle.net/6gy98n7o/), which uses jQuery 2.2.4, bootstrap 3.3.7, bootstrap-select 1.11.2.
I still can't reproduce using this minimal test case and the same Chrome/Android as my earlier comment.
I *can* reproduce it on your page.

This leads me to think it's an issue with some library or code used on your page - if you can create a minimal test case which has the problem, we can look into it.

**Comment by shehi** [comment]
Thanks a lot @birjolaxew . I will try to narrow it down to find which dep breaks it.

As you noticed, I have tweaked few CSS - I don't think that would break anything, do you? I just overrode some properties, not touching any markup this package creates in any way.

**Comment by shehi** [comment]
Guys, could someone assist me with finding the actual lines which handle this behaviour? I mean, which lines/methods handle dropdown close event when outside area is clicked, or dropdown itself is clicked again (after it was opened). I tried to debug this thing with Chrome, but to no avail. All I could find was "keydown" method, which actually doesn't get triggered when dropdowns are closed. Thanks in advance for any tips!

**Comment by shehi** [comment]
Ok, I found something strange in Bootstrap, when touch-mode of Responsive look in Firefox is enabled.

https://saucelabs.com/beta/manual/test_proxy/aeb55034c3e34aa0afd5855b2821b6d5

As you know, there is a ```clearMenus()``` function within dropdown.js of Bootstrap. That function is bound to "click" event of ```dropdown-backdrop``` element, which is created on top of the page everytime a dropdown is opened. Whenever ```dropdown-backdrop``` is clicked, function is invoked and it clears the dropdowns.

Thing is, when we simulate with Firefox's touch-enabled responsive view, after ```clearMenu()``` successfully closes all dropdowns, it is being invoked once more, this time reopening the closed dropdown. Same behavior can be observed in Chrome on Android (until recently I could observe it in Chrome-on-Desktop's mobile-UA responsive views as well, but that ceased to be the case as of late).

So question is: Why in mobile mode, all browsers invoke ```clearMenus()``` after menus get cleared? As if some missing propagation kicks in, propagates further and re-triggers the event handler.

I know this is Bootstrap related issue, but I observed this only with Bootstrap-Select. I have Bootstrap dropdowns in other contexts as well, which work quite fine.

Any ideas?
